src/Common/Core/Impl/Extensions/TaskExtensions.cs

```diff
@@ -117,6 +117,16 @@ public static class TaskExtensions {
         }
 
         /// <summary>
+        /// Waits for a task to complete within a given timeout. 
+        /// Returns task result or default(T) if task did not complete
+        /// within the timeout specified.
+        /// </summary>
+        public static T WaitTimeout<T>(this Task<T> task, int msTimeout) {
+            task.Wait(msTimeout);
+            return task.IsCompleted ? task.Result : default(T);
```

Comment:
`Result` throws `AggregateException` if there was an exception. It would be preferable to unwrap it here; or better yet, return true/false, and let the caller obtain the result as they see fit.